### PR TITLE
Fixing DNS iteration when multiple IPs are returned for the same host.

### LINKED ISSUE #549: Multipe connections detected for device in MQTT
- **OS, version, SKU and CPU architecture used:** <VERSION> 
Windows 10, Docker 18.03.1-ce-win65 linux container mode
- **SDK version used:** <VERSION> 
1.17.0

# Description of the issue:

I am trying to run an edge module in local machine (without docker). The steps are as follows
1. Start edgeHub container with all credential, network and volume. (Port is exposed correctly)
2. Open the csharp module in VSCode, and inject the connection, certificate value as environment variable
3. Run the csharp module,  error from edgeHub

 Closing AMQP device proxy for mylaptop2/target because no handler was registered.
Microsoft.Azure.Devices.Edge.Hub.Core.MultipleConnectionsException: Multiple connections detected for device mylaptop2/target

Tried to investigate, it seems the issue is caused by 
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs#L942

It creates multiple connections in one session and cause the failure 

related to the fix of issue #495

### PR DESCRIPTION
**Pull request by CIPop**
## Checklist
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [X] This pull-request is submitted against the `master` branch.

## Description of the changes

Fixing MQTT DNS iteration to stop once a viable connection was made.

## Reference/Link to the issue solved with this PR (if any)

Fixes #549

### CODE REVIEW

**Comment by CIPop** [comment]
Pending tests as described in https://github.com/Azure/azure-iot-sdk-csharp/issues/549#issuecomment-404671209 and https://github.com/Azure/azure-iot-sdk-csharp/issues/549#issuecomment-404778355

**Comment by CIPop** [comment]
Thank you all for your CR! Will resolve conflicts and send this to the integration job.

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs [hunk 1]
```diff
@@ -955,9 +956,9 @@ public override async Task SendTwinPatchAsync(TwinCollection reportedProperties,
                                 Debug.WriteLine("ConnectException trying to connect to {0}: {1}", address.ToString(), ex.ToString());
                                 return true;
                             }
+
```
Comment (adashen) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs]: the return in catch logic causes another bug. If there is error throws in the first call to ConnectAsync, there is no channel returned. We found this when trying the new SDK 1.17.1.  So it seems we need keep trying connection when exception happens

Comment (CIPop) [reply iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs]: @abhipsaMisra PTAL.

@adashen Please add more details: if the lambda passed to Handle returns `true` it means that the exception was handled and the `foreach` statement should continue (retry). If the return is false then we throw that exception.
https://msdn.microsoft.com/en-us/library/system.aggregateexception.handle(v=vs.110).aspx#Anchor_2

**Pull request merged**